src/main/java/org/folio/circulation/domain/validation/CheckInValidators.java

```diff
@@ -0,0 +1,29 @@
+package org.folio.circulation.domain.validation;
+
+import static org.folio.circulation.domain.representations.CheckInByBarcodeRequest.CLAIMED_RETURNED_RESOLVED_BY;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import org.folio.circulation.domain.CheckInProcessRecords;
+import org.folio.circulation.support.Result;
+
+public final class CheckInValidators {
+
+  private CheckInValidators() {
+  }
+
+  public static Result<CheckInProcessRecords> refuseWhenClaimedReturnedIsNotResolved(
+    Result<CheckInProcessRecords> processRecordsResult) {
+
+    return processRecordsResult.failWhen(
+      processRecords -> succeeded(isClaimedReturnedNotResolved(processRecords)),
+      processRecords -> singleValidationError(String.format(
+        "Item has Claimed returned status but no '%s' present", CLAIMED_RETURNED_RESOLVED_BY),
+        CLAIMED_RETURNED_RESOLVED_BY, null));
```

Comment:
Why is this message parameterised? If this is presented to the user, will they understand what the name of the JSON property mean?

Reply:
The property name should be self-explanatory enough. Do you have an idea how it can be reworded?

Reply:
Personally, I prefer the messages to be in domain language where possible. I might suggest something like `Item is claimed returned, a resolution is required to check in` or something like that, depending upon the language used by Emma or other domain experts.

Reply:
Changed to `Item is claimed returned, a resolution is required to check in`